common/llvm.saw

```diff
@@ -1,28 +1,51 @@
 // LLVM utility definitions
 
-// Returns a newly-allocated pointer of the given type that points to
-// the given value.
-let alloc_init ty v = do {
-    p <- crucible_alloc ty;
-    crucible_points_to p v;
-    return p;
+// Remove 'crucible' from common commands
+let points_to = crucible_points_to;
+let precond = crucible_precond;
+let from_cryptol = crucible_term;
+let execute = crucible_execute_func;
+let postcond = crucible_postcond;
+let returns = crucible_return;
+let struct = crucible_struct;
+let array = crucible_array;
+let llvm_verify module function_name overrides spec =
+  crucible_llvm_verify module function_name overrides true spec z3;
+
+// Declare constants used to denote whether an allocation is
+// 'READONLY' (CONST) or not (WRITE).
+let CONST = {{ False }};
+let WRITE = {{ True }};
+
+// Returns a newly-allocated pointer of the given type.
+let pointer (rw : Term) (type : LLVMType) = if (eval_bool {{ rw == WRITE }})
+                      then crucible_alloc type
```

Comment:
I wonder if this one should be called `alloc` because it always returns a pointer to allocated space. The `crucible_fresh_pointer` function returns an arbitrary pointer which can't be dereferenced (and may be `NULL`), and might be a better choice for abbreviation to `pointer`.